src/dds_data.cpp

```diff
@@ -209,8 +207,223 @@ QVariant CommonData::readValue(const QString& topicName,
     } // End targetMember type switch
 
     return value;
+}
+
+QVariant CommonData::readDynamicMember(const QString& topicName,
+                                       const QString& memberName,
+                                       unsigned int index)
+{
+    const char* error = "NULL";
+    QVariant value;
+    QMutexLocker locker(&m_dynamicSamplesMutex);
+
+    if (!m_dynamicSamples.contains(topicName)) {
+        return error;
+    }
+
+    const QList<DDS::DynamicData_var>& sampleList = m_dynamicSamples[topicName];
+    if (static_cast<int>(index) >= sampleList.count()) {
+        return error;
+    }
+
+    DDS::DynamicData_var sample = sampleList.at(index);
+    DDS::DynamicType_var topic_type = sample->type();
+    OpenDDS::XTypes::MemberPath member_path;
+    if (member_path.resolve_string_path(topic_type, memberName.toStdString().c_str()) != DDS::RETCODE_OK) {
+        return error;
+    }
+
+    // The direct parent dynamic data of this member
+    DDS::DynamicData_var parent_data;
+
+    // The Id of this member within the direct parent type
+    DDS::MemberId id;
+
+    if (member_path.get_member_from_data(sample, parent_data, id) != DDS::RETCODE_OK) {
+        return error;
+    }
+
+    DDS::DynamicType_var parent_type = parent_data->type();
+    DDS::DynamicTypeMember_var dtm;
+    if (parent_type->get_member(dtm, id) != DDS::RETCODE_OK) {
+        return error;
+    }
+
+    DDS::MemberDescriptor_var md;
+    if (dtm->get_descriptor(md) != DDS::RETCODE_OK) {
+        return error;
+    }
+
+    const DDS::TypeKind member_tk = md->type()->get_kind();
+    bool failed = true;
+    DDS::ReturnCode_t rc = DDS::RETCODE_OK;
+
+    switch (member_tk) {
+    case OpenDDS::XTypes::TK_BOOLEAN:
+      {
+          CORBA::Boolean tmp;
+          rc = parent_data->get_boolean_value(tmp, id);
+          if (rc == DDS::RETCODE_OK) {
+              value = tmp;
+              failed = false;
```

Comment:
Would it be simpler and clearer to just return the value here?

Reply:
Yes, I think so.